gobblin-core/src/main/java/gobblin/http/ThrottledHttpClient.java

```diff
@@ -0,0 +1,56 @@
+package gobblin.http;
+
+import java.io.IOException;
+
+import org.apache.commons.lang.exception.ExceptionUtils;
+
+import com.typesafe.config.Config;
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.broker.gobblin_scopes.GobblinScopeTypes;
+import gobblin.broker.iface.NotConfiguredException;
+import gobblin.broker.iface.SharedResourcesBroker;
+import gobblin.util.http.HttpLimiterKey;
+import gobblin.util.limiter.Limiter;
+import gobblin.util.limiter.broker.SharedLimiterFactory;
+
+
+/**
+ * A {@link HttpClient} for throttling calls to the underlying TX operation using the input
+ * {@link Limiter}.
+ */
+@Slf4j
+public abstract class ThrottledHttpClient<RQ, RP> implements HttpClient<RQ, RP>  {
+
+  protected final Limiter limiter;
+  protected final SharedResourcesBroker<GobblinScopeTypes> broker;
+  protected final Config config;
+
+  public ThrottledHttpClient (Config config, SharedResourcesBroker<GobblinScopeTypes> broker) {
+    this.config = config;
```

Comment:
We only need below configuration to create broker, this is in State not in Config.

gobblin.broker.limiter.GLOBAL.class=qps
gobblin.broker.limiter.qps=1

From config we only need urlTemplate to get the limiter key string.